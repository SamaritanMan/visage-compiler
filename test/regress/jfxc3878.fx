/*
 * @test
 * @run
 */

class sha {
  var seq1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
  var mir1 = seq1;

  function doit() {
    insert "@" before seq1[0];
    println(mir1);
    delete seq1[0..4];
    println(mir1);
    println(seq1);
  }
}

sha{}.doit();

// Let's throw in the equivalent test for script-level:

var seq2 = ['A', 'B', 'C', 'D', 'E', 'F', 'G'];
var mir2 = seq2;
insert "@" before seq2[0];
println(mir2);
delete seq2[0..4];
println(mir2);
println(seq2);

// and for local-block-level:

function f3 () {
    var seq3 = ['1', '2', '3', '4', '5', '6', '7'];
    var mir3 = seq3;
    insert "@" before seq3[0];
    println(mir3);
    delete seq3[0..4];
    println(mir3);
    println(seq3);
}
f3();
