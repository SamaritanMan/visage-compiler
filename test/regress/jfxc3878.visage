/*
 * @test
 * @run
 */

class sha1 {
  var seq1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
  var mir1 = seq1;

  function doit() {
    insert "@" before seq1[0];
    println(mir1);
    delete seq1[0..4];
    println(mir1);
    println(seq1);
  }
}

sha1{}.doit();

class sha2 {
  var seq1;
  var mir1;

  function doit() {
    seq1 = [ 'b', 'c', 'd', 'e', 'f', 'g', 'h'];
    mir1 = seq1;
    insert "@" before seq1[0];
    println(mir1);
    delete seq1[0..4];
    println(mir1);
    println(seq1);
  }
}

sha2{}.doit();

// Let's throw in the equivalent test for script-level:

var seq2 = ['A', 'B', 'C', 'D', 'E', 'F', 'G'];
var mir2 = seq2;
insert "@" before seq2[0];
println(mir2);
delete seq2[0..4];
println(mir2);
println(seq2);

// and for local-block-level:

function f3 () {
    var seq3 = ['1', '2', '3', '4', '5', '6', '7'];
    var mir3 = seq3;
    insert "@" before seq3[0];
    println(mir3);
    delete seq3[0..4];
    println(mir3);
    println(seq3);
}
f3();
