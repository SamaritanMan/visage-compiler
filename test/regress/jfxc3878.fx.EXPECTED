[ a, b, c, d, e, f, g ]
[ a, b, c, d, e, f, g ]
[ e, f, g ]
[ b, c, d, e, f, g, h ]
[ b, c, d, e, f, g, h ]
[ f, g, h ]
[ A, B, C, D, E, F, G ]
[ A, B, C, D, E, F, G ]
[ E, F, G ]
[ 1, 2, 3, 4, 5, 6, 7 ]
[ 1, 2, 3, 4, 5, 6, 7 ]
[ 5, 6, 7 ]
